lib/api/core/storage/bootstrap/safeBootstrap.js

```diff
@@ -33,28 +33,24 @@ const Bluebird = require('bluebird');
  */
 class SafeBootstrap {
   /**
-   * @param {Kuzzle} kuzzle - Kuzzle object
-   * @param {IndexEngine} indexEngine - Index engine
+   * @param {IndexStorage} indexStorage - Index storage engine
    * @param {Number} timeout - Bootstrap timeout
    */
-  constructor (kuzzle, indexEngine, timeout) {
-    this.kuzzle = kuzzle;
-    this.indexEngine = indexEngine;
-    this.attemptDelay = Math.round(timeout / 10);
+  constructor (indexStorage, timeout) {
+    this._indexStorage = indexStorage;
+    this._attemptDelay = Math.round(timeout / 10);
 
     // IDs for config documents
-    this._LOCK_ID = `${this.indexEngine.index}.lock`;
-    this._BOOTSTRAP_DONE_ID = `${this.indexEngine.index}.done`;
+    this._LOCK_ID = `${this._indexStorage.index}.lock`;
+    this._BOOTSTRAP_DONE_ID = `${this._indexStorage.index}.done`;
   }
 
   /**
    * Sequence of actions to play for bootstrap
    * @abstract
-   * @param  {...any} args - Arguments given to startOrWait()
    * @returns {Promise}
    */
-  _bootstrapSequence (...args) {
-    throw new Error(`To be implemented. args: ${args.join()}`);
+  async _bootstrapSequence () {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this function is empty or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1186)